packages/Sandblocks-Core/SBBlock.class.st

```diff
@@ -38,20 +38,20 @@ SBBlock >> absorbsInput: anEvent [
 { #category : #'action helpers' }
 SBBlock >> actionProviders [
 
-	^ {self}
+	^ {self}, self decorators
 ]
 
 { #category : #'action helpers' }
 SBBlock >> actions [
-
-	^ Array streamContents: [:stream |
+	
+	^ (Array streamContents: [:stream |
 		self actionProviders do: [:obj |
 			obj class withAllSuperclassesDo: [:class |
 				Pragma
 					withPragmasIn: class
 					do: [:pragma |
-						(#(#action #multiSelectAction #inputAction) includes: pragma keyword) ifTrue: [stream nextPut: pragma]].
-				SBBlock = class ifTrue: [^ stream contents]]]]
+						(#(#action #multiSelectAction #inputAction) includes: pragma keyword) ifTrue: [stream nextPut: pragma]]]]]) 
+		asIdentitySet asArray.
```

Comment:
ah good call, kept forgetting that I wanted to do that. Maybe use `asSet` though, shouldn't be super performance critical :)

Reply:
We tried to do that at first but for some reason, `asSet` removes most of the actions.